client/views/view_units.cpp

```diff
@@ -17,26 +18,78 @@
 #include "tileset/sprite.h"
 #include "top_bar.h"
 
+// common
+#include "canvas.h"
+#include "goto.h"
+#include "movement.h"
+#include "text.h"
+
 /**
  * Constructor for units view
  */
 units_view::units_view() : QWidget()
 {
   ui.setupUi(this);
 
+  const QPixmap *spr;
+  QImage img;
+  QRect crop;
+  QImage cropped_img;
+  QPixmap pix;
```

Comment:
Please don't get into this habit of declaring everything at the top. Just declare variables on first use, possibly using `const` and `auto`, it reduces the risk of using uninitialized variables.